Mocha's type definitions pollute and break other node modules

For example, it changes the return type of `setTimeout` and causes some errors.

```
node_modules/spica/throttle.ts(7,5): error TS2322: Type 'Timer' is not assignable to type 'number'.
node_modules/spica/throttle.ts(28,5): error TS2322: Type 'Timer' is not assignable to type 'number'.
```

This is caused by #22112.

@aozgaa Can you revert that change?

--- a/types/karma-webpack/index.d.ts
+++ b/types/karma-webpack/index.d.ts
@@ -4,9 +4,9 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.3
 
-import * as m from 'karma';
-import webpack = require('webpack');
-import webpackDevMiddleware = require('webpack-dev-middleware');
+import 'karma';
+import * as Webpack from 'webpack';
+import { Reporter, Logger } from 'webpack-dev-middleware';
 
 declare module 'karma' {
     // Note: karma-webpack will set publicPath for us, so it is optional here.
@@ -16,24 +16,24 @@ declare module 'karma' {
         noInfo?: boolean;
         quiet?: boolean;
         lazy?: boolean;
-        watchOptions?: webpack.Options.WatchOptions;
+        watchOptions?: Webpack.Options.WatchOptions;
         publicPath?: string;
         index?: string;
         headers?: {
             [name: string]: string;
         };
-        stats?: webpack.Options.Stats;
-        reporter?: webpackDevMiddleware.Reporter | null;
+        stats?: Webpack.Options.Stats;
+        reporter?: Reporter | null;
         serverSideRender?: boolean;
 
-        log?: webpackDevMiddleware.Logger;
-        warn?: webpackDevMiddleware.Logger;
-        error?: webpackDevMiddleware.Logger;
+        log?: Logger;
+        warn?: Logger;
+        error?: Logger;
         filename?: string;
     }
 
     interface ConfigOptions {
-        webpack: webpack.Configuration;
+        webpack: Webpack.Configuration;
         webpackMiddlewareOptions: KarmaWebpackMiddlewareOptions;
     }
 }

--- a/types/koa-webpack/koa-webpack-tests.ts
+++ b/types/koa-webpack/koa-webpack-tests.ts
@@ -12,8 +12,6 @@ const middleware = koaWebpack({
     compiler,
     config,
     dev: {
-        noInfo: false,
-        quiet: false,
         lazy: true,
         watchOptions: {
             aggregateTimeout: 300,

--- a/types/mocha/index.d.ts
+++ b/types/mocha/index.d.ts
@@ -3,8 +3,6 @@
 // Definitions by: Kazi Manzur Rashid <https://github.com/kazimanzurrashid>, otiai10 <https://github.com/otiai10>, jt000 <https://github.com/jt000>, Vadim Macagon <https://github.com/enlight>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-/// <reference types="node" />
-
 interface MochaSetupOptions {
     //milliseconds to wait before considering a test slow
     slow?: number;
@@ -174,7 +172,7 @@ declare namespace Mocha {
 
 
     /** Partial interface for Mocha's `Runner` class. */
-    interface IRunner extends NodeJS.EventEmitter { }
+    interface IRunner { }
 
     interface IContextDefinition {
         (description: string, callback: (this: ISuiteCallbackContext) => void): ISuite;

--- a/types/mocha/mocha-tests.ts
+++ b/types/mocha/mocha-tests.ts
@@ -455,9 +455,3 @@ function test_run_withOnComplete() {
 function test_throwError() {
     mocha.throwError(new Error("I'm an error!"));
 }
-
-function test_runner_fluentParams() {
-    new MochaDef().run(); // $ExpectType IRunner
-    new MochaDef().run().on("event-string", function() {}); // $ExpectType IRunner
-    new MochaDef().run().on(Symbol("event-symbol"), function() {}); // $ExpectType IRunner
-}

--- a/types/webpack-dev-middleware/index.d.ts
+++ b/types/webpack-dev-middleware/index.d.ts
@@ -5,9 +5,9 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.2
 
-import { NextHandleFunction } from 'connect';
-import { Logger } from 'loglevel';
 import * as webpack from 'webpack';
+import * as loglevel from 'loglevel';
+import { NextHandleFunction } from 'connect';
 import MemoryFileSystem = require('memory-fs');
 
 export = WebpackDevMiddleware;
@@ -40,6 +40,7 @@ declare namespace WebpackDevMiddleware {
 		log: Logger;
 	}
 
+	type Logger = loglevel.Logger;
 	type Reporter = (middlewareOptions: Options, reporterOptions: ReporterOptions) => void;
 
 	interface WebpackDevMiddleware {

